test/functional/feature_framework_unit_tests.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2024 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Framework unit tests
+
+Unit tests for the test framework.
+"""
+
+import sys
+import io
+import unittest
+
+from test_framework.test_framework import TEST_EXIT_PASSED, TEST_EXIT_FAILED
+
+# List of framework modules containing unit tests. Should be kept in sync with
+# the output of `git grep unittest.TestCase ./test/functional/test_framework`
+TEST_FRAMEWORK_MODULES = [
+    "address",
+    "crypto.bip324_cipher",
+    "blocktools",
+    "crypto.chacha20",
+    "crypto.ellswift",
+    "key",
+    "messages",
+    "crypto.muhash",
+    "crypto.poly1305",
+    "crypto.ripemd160",
+    "script",
+    "segwit_addr",
+]
+
+
+def run_unit_tests():
+    test_framework_tests = unittest.TestSuite()
+    for module in TEST_FRAMEWORK_MODULES:
+        test_framework_tests.addTest(
+            unittest.TestLoader().loadTestsFromName(f"test_framework.{module}")
+        )
+    with io.StringIO() as buf:
+        result = unittest.TextTestRunner(stream=buf, verbosity=1, failfast=True).run(
+            test_framework_tests
+        )
+        if not result.wasSuccessful():
+            print(f"{buf.getvalue()}", file=sys.stderr)
+            sys.exit(TEST_EXIT_FAILED)
+        sys.exit(TEST_EXIT_PASSED)
```

Comment:
Thank you for reviewing.  **Would you be ok with a simplified approach of writing all unit test output to stdout, with this output being shown when unit tests fail?**  This seems a bit cleaner and more consistent than selectively placing all unit test output to stdout or stderr depending on test pass/fail.


### More info
`TextTestRunner` lets us print unit test output to one stream.  Previously (before this PR), `TextTestRunner` wrote all output to stderr regardless of pass/fail but doing this now would cause a false failure (test runner interprets any stderr to mean the test failed).

As was the case before this PR, unit test pass/fail would continue to be based on `TestResult` returned.

As you mentioned, stdout is hidden by the test runner in non-failure situations, so writing to stdout on test pass doesn't do much currently (other than providing the output to test runner for future use).  The user is provided with a reasonable test pass/fail status (matching that of the other tests).

### Example output
#### Pre-PR (pass)
```
$ test/functional/test_runner.py --jobs=18 --cachedir=/mnt/tmp/cache --tmpdir=/mnt/tmp test/functional/feature_signet.py
Temporary test directory at /mnt/tmp/test_runner_₿_🏃_20240403_192403
Running Unit Tests for Test Framework Modules
.....................
----------------------------------------------------------------------
Ran 21 tests in 8.262s

OK
Remaining jobs: [feature_signet.py]
1/1 - feature_signet.py passed, Duration: 8 s

TEST              | STATUS    | DURATION

feature_signet.py | ✓ Passed  | 8 s

ALL               | ✓ Passed  | 8 s (accumulated) 
Runtime: 8 s
```
#### Pre-PR (failure in unit test)
```
$ test/functional/test_runner.py --jobs=18 --cachedir=/mnt/tmp/cache --tmpdir=/mnt/tmp test/functional/feature_signet.py
Temporary test directory at /mnt/tmp/test_runner_₿_🏃_20240403_192509
Running Unit Tests for Test Framework Modules
.F
======================================================================
FAIL: test_bech32_decode (test_framework.address.TestFrameworkScript.test_bech32_decode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dev/bitcoin/test/functional/test_framework/address.py", line 228, in test_bech32_decode
    self.assertEqual(False, True)
AssertionError: False != True

----------------------------------------------------------------------
Ran 2 tests in 0.003s

FAILED (failures=1)
Early exiting after failure in TestFramework unit tests
```

#### PR (pass, running unit and functional tests):
```
$ test/functional/test_runner.py --jobs=18 --cachedir=/mnt/tmp/cache --tmpdir=/mnt/tmp test/functional/feature_signet.py test/functional/feature_framework_unit_tests.py 
Temporary test directory at /mnt/tmp/test_runner_₿_🏃_20240403_193148
Remaining jobs: [feature_signet.py, feature_framework_unit_tests.py]
1/2 - feature_signet.py passed, Duration: 8 s
Remaining jobs: [feature_framework_unit_tests.py]
2/2 - feature_framework_unit_tests.py passed, Duration: 9 s

TEST                            | STATUS    | DURATION

feature_framework_unit_tests.py | ✓ Passed  | 9 s
feature_signet.py               | ✓ Passed  | 8 s

ALL                             | ✓ Passed  | 17 s (accumulated) 
Runtime: 9 s
```


#### PR (pass, running only unit tests):
```
$ test/functional/test_runner.py --jobs=18 --cachedir=/mnt/tmp/cache --tmpdir=/mnt/tmp test/functional/feature_framework_unit_tests.py
Temporary test directory at /mnt/tmp/test_runner_₿_🏃_20240403_193240
Remaining jobs: [feature_framework_unit_tests.py]
1/1 - feature_framework_unit_tests.py passed, Duration: 9 s

TEST                            | STATUS    | DURATION

feature_framework_unit_tests.py | ✓ Passed  | 9 s

ALL                             | ✓ Passed  | 9 s (accumulated) 
Runtime: 9 s
```

#### PR (failure in unit tests):
```
$ test/functional/test_runner.py --jobs=18 --cachedir=/mnt/tmp/cache --tmpdir=/mnt/tmp test/functional/feature_framework_unit_tests.py
Temporary test directory at /mnt/tmp/test_runner_₿_🏃_20240403_193325
Remaining jobs: [feature_framework_unit_tests.py]
1/1 - feature_framework_unit_tests.py failed, Duration: 0 s

stdout:
.F
======================================================================
FAIL: test_bech32_decode (test_framework.address.TestFrameworkScript.test_bech32_decode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dev/myrepos/bitcoin/test/functional/test_framework/address.py", line 228, in test_bech32_decode
    self.assertEqual(False, True)
AssertionError: False != True

----------------------------------------------------------------------
Ran 2 tests in 0.003s

FAILED (failures=1)

stderr:


TEST                            | STATUS    | DURATION

feature_framework_unit_tests.py | ✖ Failed  | 0 s

ALL                             | ✖ Failed  | 0 s (accumulated) 
Runtime: 0 s
```